### PR TITLE
Allow pose_estimator to consume INS state

### PR DESCRIPTION
**Pull request by hidmic**
# Description

As per offline discussion, the `pose_estimator` should be able to consume either INS or AHRS data to provide an equally good or better state estimate. This patch adapts the former to do this. Additionally, synchronization primitives introduced in #91 are dropped and replaced by a simpler scheme where state publications are tied to one input: either the INS or the AHRS.

## Type of change

Please mark options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] The code builds clean without any errors or warnings
- [ ] Unit tests are passing
- [x] Linter tests are passing

# How To Test

Perform full manual QA with and without an INS.

### CODE REVIEW

File: pose_estimator/src/pose_estimator.cpp [hunk 1]
```diff
@@ -278,22 +285,28 @@ void PoseEstimator::dataCallback(
...
-  // Estimate velocity from AHRS and thruster RPMs
-  const double speed_kn = static_cast<double>(rpms_msg->rpms) / rpm_per_kn_;
-  constexpr double kn_per_ms = 1852. / 3600.;  // exact conversion
-  const tf::Quaternion nominal_velocity(speed_kn * kn_per_ms, 0., 0., 0.);
-  const tf::Quaternion velocity =
+  if (last_rpms_msg_)
+  {
+    // Estimate velocity from AHRS and thruster RPMs
+    const double speed_kn = static_cast<double>(last_rpms_msg_->rpms) / rpm_per_kn_;
+    constexpr double kn_per_ms = 1852. / 3600.;  // exact conversion
```
Comment (patapufate) [inline pose_estimator/src/pose_estimator.cpp:302]: where this number come from?

**Review by patapufate — APPROVED** [review]
LGTM

File: pose_estimator/src/pose_estimator.cpp (hunk 1 above)
Comment (hidmic) [reply pose_estimator/src/pose_estimator.cpp:302]: AFAIK 1.852 km/h is the official SI equivalent for nautical knots. I cannot find official documentation, hope [Wikipedia is enough](https://en.wikipedia.org/wiki/Knot_(unit)).

**Comment by hidmic** [comment]
Rebased to bring in all recent changes.

**Comment by hidmic** [comment]
Tested on the vehicle. Appears to be working correctly.

File: pose_estimator/src/pose_estimator.cpp (hunk 1 above)
Comment (patapufate) [reply pose_estimator/src/pose_estimator.cpp:302]: SGTM

**Pull request merged**